pulsar-client/src/main/java/org/apache/pulsar/client/impl/MemoryLimitController.java

```diff
@@ -45,11 +60,18 @@ public boolean tryReserveMemory(long size) {
             }
 
             if (currentUsage.compareAndSet(current, newUsage)) {
+                checkTrigger(current, newUsage);
                 return true;
             }
         }
     }
 
+    private void checkTrigger(long prevUsage, long newUsage) {
+        if (newUsage >= triggerThreshold && prevUsage < triggerThreshold && trigger != null) {
```

Comment:
Would this be better?
`if (trigger != null && newUsage >= triggerThreshold && prevUsage < triggerThreshold) `

Reply:
I put this at the last on purpose. `trigger` is null only in unit test.